efabless/EF_UVM
# Add CI to run EF_UVM test for some IPs

Adds a CI that does the following:
1. Takes `.github/test_set.yaml` as an input:
```
- name: EF_GPIO8
  url: https://github.com/efabless/EF_GPIO8.git
  tests:
  - gpio8_all_out_test 
  - gpio8_all_in_test
```
2. Creates a test matrix based on the file above where each entry in the matrix represents a specific test for a specific IP.
3. Clones IPs and run tests. Each entry in the matrix runs in a separate job. 

The purpose of this CI is to try and verify if changes in EF_UVM affects exiting tests. 

Future Enhancements (for this PR or others):
- Add more IPs and more IP tests.
- Save EF_UVM commit hash and IP commit hash for reproducibility.   

Note:
- Please use squash and merge option.
